src/concepts/rendering-under-the-hood.md

```diff
@@ -0,0 +1,265 @@
+        Paragraph::new("Hello World!"),
+        frame.size(),
+    );
+}
+```
+
+This `Frame` struct contains a method called `render_widget` which takes any object that implements
+the `Widget` trait. This `render_widget` method calls the `Widget::render` method on the type-erased
+`Widget` struct.
+
+```rust
+    pub fn render_widget<W>(&mut self, widget: W, area: Rect)
+    where
+        W: Widget,
+    {
+        widget.render(area, self.buffer);
+    }
+```
+
+As mentioned above, implementing the `Widget` trait is the primary way to make a struct renderable.
+As an example, here's the full implementation for the `Clear` widget:
+
+```rust
+pub struct Clear;
+
+impl Widget for Clear {
+    fn render(self, area: Rect, buf: &mut Buffer) {
+        for x in area.left()..area.right() {
+            for y in area.top()..area.bottom() {
+                buf.get_mut(x, y).reset();
+            }
+        }
+    }
+}
+```
+
+There are 2 things that you should notice.
+
+1. The `Widget` gets the current area as a `Rect`.
+2. The `Widget` gets a mutable reference to a "`Buffer`".
+
+The top level widget usually gets a reference to the entire size of the terminal as a `Rect`, i.e.
+`f.size()`. However, nested widgets may get a smaller area. Any implementation of a `Widget` must
+only draw "within the lines" of the `Rect` area.
+
+You might think that a `Widget` renders directly to the terminal but that is not the case. Ratatui
+uses something called a "double-buffer" rendering technique.
+
+Every time `terminal.draw(|f| ...)` is called, a new [`Frame`] struct is constructed that contains a
+mutable reference to an instance of the `Buffer` struct:
+
+```rust
+pub struct Frame<'a> {
+    // --snip--
+    /// The buffer that is used to draw the current frame
+    buffer: &'a mut Buffer,
+}
+```
+
+The `Buffer` represents the area in which content is going to be drawn to the terminal. The `Buffer`
+struct implements a number of methods such as [`get_mut`] and [`set_string`]:
+
+```rust
+pub fn get_mut(&mut self, x: u16, y: u16) -> &mut Cell {
+}
+
+pub fn set_string<S>(&mut self, x: u16, y: u16, string: S, style: Style)
+where
+    S: AsRef<str> {
+}
+```
+
+These methods allow any implementation of the `Widget` trait to write into any parts of the
+`Buffer`.
+
+A [`Buffer`] is essentially a collection of `Cell`s:
+
+```rust
+pub struct Buffer {
+    // --snip--
+    pub content: Vec<Cell>,
+}
+```
+
+In Ratatui, this `Cell` struct is the smallest renderable unit of code. A `Cell` essentially has a
+`String` which is the symbol that will be drawn in that `Cell` as well as `Style` information
+(foreground color, background color, modifiers etc).
+
+```rust
+// pseudo code
+pub struct Cell {
+  content: String,
+  style: Style,
+}
+```
+
+```admonish
+`Cell`s are the closest analog to a "pixel" in a terminal.
+Most terminals render monospaced text 2 high and 1 wide.
+A `Cell` in Ratatui should usually contain 1 wide string content.
+```
```

Comment:
drop the admonish?

```suggestion
`Cell`s are similar to a "pixel" in a graphical UI, bu instead of storing just a single point with color information
they store a single character. Terminals generally render text so that each individual cell takes up space
approximately twice as high as it is wide.
```

Add an admonish for Emoji (and perhaps link to a section)

Reply:
Skipping Emoji / unicode discussion for now until we add a separate page for it. Once I do, I'll update this page with a link to it.